coucharchive

```diff
@@ -255,6 +257,35 @@ class ReplicationControl(object):
         self._last_successes = multiprocessing.Manager().list()
         self._last_errors = multiprocessing.Manager().list()
 
+        self.has_run_queue_control = has_run_queue_control
+        if self.has_run_queue_control:
+            def run_queue_control():
+                t = threading.currentThread()
+                while getattr(t, "do_run", True):
+                    time.sleep(2)  # arbitrary
```

Comment:
This line would be better at the end of the loop, otherwise, when we decide to stop, the thread will still be "survive-waiting" for 2 seconds.

Reply:
OK.